### PR TITLE
Fix awful.keygrabber:add_keybinding function

### LINKED ISSUE #3567: Error with add_keygrabber: table expected, got nil
(on awesome master)

I write this simple keygrabber:
```
local exit_screen_grabber = awful.keygrabber {
    keybindings = {
        awful.key { key = 'Escape',
                    on_press = exit_screen_hide}
    },
}
```
Then I try to add a keybinding:
```
exit_screen_grabber:add_keybinding(
    awful.key {
        key = "x",
        on_press = action
})
```
But I get this weird error:
```
stack traceback:
        [C]: in function 'table.insert'
        /usr/share/awesome/lib/awful/keygrabber.lua:682: in method 'add_keybinding'
        /home/grumph/.config/awesome/widgets/exit_screen.lua:130: in local 'big_button_widget'
        /home/grumph/.config/awesome/widgets/exit_screen.lua:165: in main chunk
        [C]: in function 'require'
        /home/grumph/.config/awesome/rc.lua:68: in main chunk
error: /usr/share/awesome/lib/awful/keygrabber.lua:682: bad argument #1 to 'insert' (table expected, got nil)
```

Then I try to read the code of `add_keygrabber` and I don't understand:

https://github.com/awesomeWM/awesome/blob/e26069682635b734d61c251e261d8fcadcd7522c/lib/awful/keygrabber.lua#L530-L531

In both cases, `key` is an awful.key object and `_keycode` is a string.

I fixed my problem by replacing the first line with:
```
    self._private.keybindings[_keycode] = self._private.keybindings[_keycode] or {}
```

Bonus:
I find this line suspicious too

https://github.com/awesomeWM/awesome/blob/e26069682635b734d61c251e261d8fcadcd7522c/lib/awful/keygrabber.lua#L512

I think it should be
```
local mods = not key._is_awful_key and key or nil
```

### PR DESCRIPTION
**Pull request by Grumph**
Fix #3567

### CODE REVIEW

**Review by Aire-One — COMMENTED** [review]
Nice catch! Thank you, @Grumph.

I'm not so sure about the test, tho... The only useful part of the test is the call to `add_keybinding`, and I guess it should be in some unit test rather than these integration tests that were done to a different use case (see top file comment).

**Comment by Aire-One** [comment]
I tried to write the unit test myself. For some reason I can't PR your fork of awesome, so my attempt can only be seen in my repository at https://github.com/Aire-One/awesome/commit/ab0a7ad6d28c37815ea80dd2e3e2e9bd0a1c276a

**Comment by Grumph** [comment]
> I'm not so sure about the test

Sorry for that, I didn't take time to understand where was what, codecov raised an alert so I just went for a dumb `find . -iname '*test*'` + copy/paste other tests :grin: 
As `add_keybinding` failed before my patch I thought it would to the job.

Shall I revert 808b17cd5c5a7c6dd95ac71b5cb3167775494a95 ?

**Review by Aire-One — APPROVED** [review]
Thank you for the update :)

File: spec/preload.lua [hunk 1]
```diff
@@ -1,12 +1,15 @@
 -- This script is given to Busted via the --helper argument. Modules loaded here
 -- won't be cleared and reloaded by Busted. This is needed for lgi because lgi
 -- is not safe to reload and yet Busted manages to do this.
-require("lgi")
+require "lgi"
```
Comment (actionless) [inline spec/preload.lua:4]: any reason for this?

File: spec/preload.lua [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 -- Always show deprecated messages
-_G.awesome = {version = "v9999"}
+_G.awesome = {
+   version = "v9999",
+   api_level = 9999,
+}
 
 -- "fix" some intentional beautiful breakage done by .travis.yml
-require("beautiful").init{a_key="a_value"}
+require("beautiful").init { a_key = "a_value" }
```
Comment (actionless) [inline spec/preload.lua:13]: this formatting change in a totally random place and this space in `init {` doesnt match the rest of codebase

**Comment by actionless** [comment]
thanks for also adding the tests 👍

File: spec/preload.lua (hunk 1 above)
Comment (Aire-One) [reply spec/preload.lua:4]: No particular reason. I think it's from my formatter and I was too lazy to not git-commit it.

Comment (actionless) [reply spec/preload.lua:4]: let's just try to keep it uniform

**Review by Elv13 — APPROVED** [review]

**Pull request merged**